var/spack/repos/builtin/packages/py-scikit-build-core/package.py

```diff
@@ -0,0 +1,49 @@
+# Copyright 2013-2023 Lawrence Livermore National Security, LLC and other
+# Spack Project Developers. See the top-level COPYRIGHT file for details.
+#
+# SPDX-License-Identifier: (Apache-2.0 OR MIT)
+
+from spack.package import *
+
+
+class PyScikitBuildCore(PythonPackage):
+    """scikit-build-core is a doubly improved build system generator
+    for CPython C/C++/Fortran/Cython extensions. It features several
+    improvements over the classic scikit-build build system generator."""
+
+    homepage = "https://github.com/scikit-build/scikit-build-core"
+    pypi = "scikit_build_core/scikit_build_core-0.2.0.tar.gz"
+    git = "https://github.com/scikit-build/scikit-build-core"
+
+    maintainers("wdconinc")
+
+    version("0.2.0", sha256="d2a76d9447a412038dc5e25dd259b03c25278661a0c7c3da766bb971c1a9acd2")
+
+    depends_on("python@3.7:", type=("build", "run"))
+    depends_on("py-hatchling", type="build")
+    depends_on("py-hatch-vcs", type="build")
+    depends_on("py-packaging@20.9:", type=("build", "run"))
+    depends_on("py-pathspec@0.10.1:", type=("build", "run", "test"))
+    depends_on("py-pyproject-metadata@0.5:", type=("build", "run", "test"))
```

Comment:
These deps are optional, you should add a `+pyproject` variant to control these

Reply:
Variant added.